covid19_sfbayarea/data/san_francisco.py

```diff
@@ -168,67 +173,65 @@ def get_age_table(session : SocrataApi, resource_ids: Dict[str, str]) -> List[Di
-        raise FormatError(
-            f"Did not get expected Gender categories. Data received: {data}")
-
-    # re-key
-    gender_table = {GENDER_KEYS[entry["gender"]]: int(
-        entry["cumul_cases"]) for entry in data}
-
-    return gender_table
+    GENDER_KEYS = {"Female": "female", "Male": "male", "Unknown": "unknown", "Trans Female": "female", "Trans Male": "male"}
+    # find the latest date of data collection
+    params = {'$select': 'max(specimen_collection_date) as date'}
+    latest_date = session.resource(resource_id, params=params)[0]
 
-def get_transmission_table(session : SocrataApi, resource_ids: Dict[str, str]) -> Dict:
-    """Get cases by transmission category"""
-    resource_id = resource_ids['cases_deaths_transmission']
-    # Dict of source_label:target_label for re-keying
-    TRANSMISSION_KEYS = {"Community": "community",
-                        "From Contact": "from_contact", "Unknown": "unknown"}
-    params = { '$select': 'transmission_category, sum(case_count)', '$group': 'transmission_category'}
+    # get cumulative confirmed cases on latest date
+    params = {'$select': 'gender, cumulative_confirmed_cases as cases', '$where':f'specimen_collection_date="{latest_date["date"]}"'}
     data = session.resource(resource_id, params=params)
+
     # re-key
-    transmission_data = { TRANSMISSION_KEYS[ entry["transmission_category"] ]: int(entry["sum_case_count"]) for entry in data}
-    return transmission_data
+    table : Dict[str, int]= dict()
+    for entry in data:
+        rekey = GENDER_KEYS[entry['gender']]
+        table[rekey] = table.get(rekey,0) + int(entry["cases"])
+    return table
 
 # Confirmed cases by race and ethnicity
 def get_race_eth_table(session: SocrataApi, resource_ids: Dict[str, str]) -> Dict:
     """
     Fetch race x ethnicity data. Individuals are assigned to one race/eth category.
-    Individuals identified as 'Hispanic or Latino' are assigned 'Latinx_or_Hispanic'.
-    Individuals identified as 'Not Hispanic or Latino' are assigned to their race identification.
-    Due to an error in the source data, it appears that Native American datapoint is not currently assigned a race label.
-    TO-DO: update this code when the Native American cases are assigned a race label.
     """
-    resource_id = resource_ids["race_eth"]
+    resource_id = resource_ids['race_eth']
     # Dict of source_label:target_label for re-keying.
-
     RACE_ETH_KEYS = {'Hispanic or Latino/a, all races': 'Latinx_or_Hispanic', 'Asian': 'Asian', 'Black or African American': 'African_Amer', 'White': 'White',
                     'Native Hawaiian or Other Pacific Islander': 'Pacific_Islander', 'Native American': 'Native_Amer', 'Multi-racial': 'Multiple_Race', 'Other': 'Other', 'Unknown': 'Unknown'}
+    # find the latest date of data collection
+    params = {'$select': 'max(specimen_collection_date) as date'}
+    latest_date = session.resource(resource_id, params=params)[0]
```

Comment:
Same as above w/r/t extracting the date.

```suggestion
    latest_date = session.resource(resource_id, params=params)[0]['date']
```

Actually, given how much you’re repeating this, is it worth abstracting into a function you can call from all the places you do it?

```py
def get_latest_specimen_date(resource_id: str) -> str:
    data = session.resource(resource_id, params={
        '$select': 'max(specimen_collection_date)'
    })
    return data[0]['date']
```